includes/multi-currency/Compatibility/WooCommerceSubscriptions.php

```diff
@@ -143,56 +145,36 @@ public function maybe_disable_mixed_cart( $value ) {
 
-		$subscription_renewal = $this->cart_contains_renewal();
-		if ( $subscription_renewal ) {
-			$order = wc_get_order( $subscription_renewal['subscription_renewal']['renewal_order_id'] );
-			return $order ? $order->get_currency() : $return;
-		}
+		// Loop through subscription types and check for cart items.
+		foreach ( self::SUBSCRIPTION_TYPES as $type ) {
+			$cart_item = $this->get_subscription_type_from_cart( $type );
+			if ( $cart_item ) {
+				$this->running_override_selected_currency_filters = true;
 
-		// The running_override_selected_currency_filters property has been added here due to if it isn't, it will create an infinite loop of calls.
-		if ( isset( WC()->session ) && WC()->session->get( 'order_awaiting_payment' ) ) {
-			$this->running_override_selected_currency_filters = true;
-			$order = wc_get_order( WC()->session->get( 'order_awaiting_payment' ) );
-			$this->running_override_selected_currency_filters = false;
-			if ( $order && $this->order_contains_renewal( $order ) ) {
-				return $order->get_currency();
-			}
-		}
+				// If we have a cart item, then we can get the order or subscription to pull the currency from.
+				$subscription_type = 'subscription_' . $type;
+				$subscription      = $this->get_subscription( $cart_item[ $subscription_type ]['subscription_id'] );
 
-		// The running_override_selected_currency_filters property is used to avoid an infinite loop
-		// that can occur on the product page when `get_subscription()` is used.
-		$switch_id = $this->get_subscription_switch_id_from_superglobal();
-		if ( $switch_id ) {
-			$this->running_override_selected_currency_filters = true;
-			$switch_subscription                              = $this->get_subscription( $switch_id );
-			$this->running_override_selected_currency_filters = false;
-			return $switch_subscription ? $switch_subscription->get_currency() : $return;
-		}
-
-		$switch_cart_items = $this->get_subscription_switch_cart_items();
-		if ( 0 < count( $switch_cart_items ) ) {
-			$switch_cart_item    = array_shift( $switch_cart_items );
-			$switch_subscription = $this->get_subscription( $switch_cart_item['subscription_switch']['subscription_id'] );
-			return $switch_subscription ? $switch_subscription->get_currency() : $return;
-		}
-
-		$subscription_resubscribe = $this->cart_contains_resubscribe();
-		if ( $subscription_resubscribe ) {
-			$subscription = $this->get_subscription( $subscription_resubscribe['subscription_resubscribe']['subscription_id'] );
-			return $subscription ? $subscription->get_currency() : $return;
+				$this->running_override_selected_currency_filters = false;
+				return $subscription ? $subscription->get_currency() : $return;
+			}
 		}
 
-		return $return;
+		// This instance is for when the customer lands on the product page to choose a new subscription tier.
+		$switch_subscription = $this->get_subscription_from_superglobal_switch_id();
+		return $switch_subscription ? $switch_subscription->get_currency() : $return;
```

Comment:
If this needs to be written more clearly, let me know. Basically, it tries to get a sub, and if the sub exists, get its currency, otherwise return what was initially passed.

Reply:
It makes sense to me.